webapp/sources/ldap-inventory/inventory-repository/src/main/scala/com/normation/inventory/ldap/core/FullInventoryRepositoryImpl.scala

```diff
@@ -337,6 +338,79 @@ class FullInventoryRepositoryImpl(
     }
   }.chainError(s"Error when getting all node inventories for status '${inventoryStatus.name}'")
 
+  override def getInventories(inventoryStatus: InventoryStatus, nodeIds: Set[NodeId]): IOResult[Map[NodeId, FullInventory]] = {
+    /*
+     * We need to only get back the tree for nodes that we are looking for, we need to build filter like:
+     * "(entryDN:dnSubtreeMatch:=cn=group_a,dc=abc,dc=xyz)"
+     */
+    def buildSubTreeFilter[A](base: DN, ids: List[A], dnToString: A => String): Filter = {
+      val dnFilters = Filter.create(s"entryDN=${base.toString()}") :: ids
+        .map(dnToString)
+        .sorted
+        .map(a => Filter.create(s"entryDN:dnSubtreeMatch:=${a}"))
+      BuildFilter.OR(dnFilters: _*)
+    }
+
+    for {
+      con      <- ldap
+      dit       = inventoryDitService.getDit(inventoryStatus)
+      // Get base tree, we will go into each subtree after
+      nodeTree <- con
```

Comment:
there are several nodes, so naming it nodesTree would help maintenance

Reply:
It was to keep the same structure than above method, but the smart move would have been to factor out that actually